clang/test/CodeGenSYCL/sycl-device.cpp

```diff
@@ -3,6 +3,78 @@
 
 int bar(int b);
 
+int bar10(int a) { return a + 10; }
+int bar20(int a) { return a + 20; }
+
+class A {
+public:
+  // CHECK-DAG: define linkonce_odr spir_func void @_ZN1A3fooEv
+  // CHECK-DAG: define spir_func i32 @_Z5bar20i
+  __attribute__((sycl_device)) void foo() { bar20(10); }
+
+  // CHECK-DAG: define linkonce_odr spir_func void @_ZN1AC1Ev
+  __attribute__((sycl_device))
+  A() { bar10(10); }
+  // CHECK-DAG: define spir_func i32 @_Z5bar10i
+  // CHECK-DAG: define linkonce_odr spir_func void @_ZN1AD1Ev
+  __attribute__((sycl_device)) ~A() {}
+
+  template <typename T>
+  __attribute__((sycl_device)) void AFoo(T t) {}
+
```

Comment:
Can we have methods w/o `sycl_device` attribute?
Can they be called from the annotated member functions?